examples/gaussian_process/plot_gpr_prior_posterior.py

```diff
@@ -3,76 +3,259 @@
 Illustration of prior and posterior Gaussian process for different kernels
 ==========================================================================
 
-This example illustrates the prior and posterior of a GPR with different
-kernels. Mean, standard deviation, and 10 samples are shown for both prior
-and posterior.
+This example illustrates the prior and posterior of a
+:class:`~sklearn.gaussian_process.GaussianProcessRegressor` with different
+kernels. Mean, standard deviation, and 5 samples are shown for both prior
+and posterior distributions.
+
+Here, we only give some illustration. To know more about kernels' formulation,
+refer to the :ref:`User Guide <gp_kernels>`.
 """
 print(__doc__)
 
 # Authors: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
-#
+#          Guillaume Lemaitre <g.lemaitre58@gmail.com>
 # License: BSD 3 clause
 
+# %%
+# Helper function
+# ---------------
+#
+# Before to present each individual kernel available for Gaussian processes,
+# we will define an helper function to help us plotting samples drawn from
+# the Gaussian process.
+#
+# This function will take a
+# :class:`~sklearn.gaussian_process.GaussianProcessRegressor` model and will
+# drawn sample from the Gaussian process. If the model was not fit, the samples
+# are drawn from the prior distribution while after fit, the samples are drawn
+# from the posterior distribution.
+import matplotlib.pyplot as plt
 import numpy as np
 
-from matplotlib import pyplot as plt
 
+def plot_gpr_samples(gpr_model, n_samples, ax):
+    """Plot samples drawn from the Gaussian process model.
+
+    If the Gaussian process model is not train then the drawn samples are
+    drawn from the prior distributions. Otherwise, the samples are drawn from
+    the posterior distribution. Be aware that a sample here correspond to a
```

Comment:
I think it might be useful to recall why samples correspond to functions here or before.

If there's a resource within the docs, I think a link is sufficient.